### DISCUSSION TITLE
As of version 17 (JavaCPP presets version 1.5.10), `libLLVM.so` now depends on `libtinfo.so.6`

### DISCUSSION THREAD

**Discussion by 0x6675636b796f75676974687562** [bytedeco/javacpp-presets#1537] [Q&A] [answered]
Hello Team!

We've recently upgraded LLVM from version **16** to version **17**, to find that it now depends on `libtinfo.so.6` (from ncurses **6.x**):

```console
$ ldd ~/.javacpp/cache/llvm-17.0.6-1.5.10-linux-x86_64.jar/org/bytedeco/llvm/linux-x86_64/libLLVM-17.so
        linux-vdso.so.1 (0x00007fffde7ab000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f36112ad000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f36112a8000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f36112a3000)
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f3611284000)
        libtinfo.so.6 => /lib/x86_64-linux-gnu/libtinfo.so.6 (0x00007f3611251000)
        libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f3611035000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f3610f56000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f3610f36000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f3610d55000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f3619785000)
```

What's the purpose of the terminal info library here?

Is it really necessary?

Looking at the undefined (external) symbols in `libLLVM-17.so`, there're indeed 4 new ones from `libtinfo.so.6`:

```console
$ grep -F TINFO libLLVM-17.so.txt
                 U del_curterm@@NCURSES6_TINFO_5.0.19991023
                 U set_curterm@@NCURSES6_TINFO_5.0.19991023
                 U setupterm@@NCURSES6_TINFO_5.0.19991023
                 U tigetnum@@NCURSES6_TINFO_5.0.19991023
```

Yet, I can't seem to find where exactly they're used in the original code at https://github.com/llvm/llvm-project.

**Comment by saudet** [comment, marked as answer]
No reason in particular. It just looks like recent versions of LLVM enable that option by default, whereas older versions did not. I've disabled it in commit 3af42f8f5b9c3cc47f91c6ffb7c310be3db1814f. Please give it a try with the snapshots: http://bytedeco.org/builds/

**Reply by 0x6675636b796f75676974687562** [reply]
Samuel @saudet, many thanks for your response as well as the fix.

One more question about the upgrade to Clang **17** (presets **1.5.10**).

The previous version (**16**, presets **1.5.9**) required Glibc 2.16:

```
                 U aligned_alloc@@GLIBC_2.16
                 U memcpy@@GLIBC_2.14
                 U posix_spawn@@GLIBC_2.15
                 U pthread_getname_np@@GLIBC_2.12
                 U pthread_setname_np@@GLIBC_2.12
```

The new version needs Glibc 2.29, released in 2019:

```
                 U exp@@GLIBC_2.29
                 U log2f@@GLIBC_2.27
                 U log2@@GLIBC_2.29
                 U log@@GLIBC_2.29
                 U __longjmp_chk@@GLIBC_2.11
                 U memcpy@@GLIBC_2.14
                 U posix_spawn@@GLIBC_2.15
                 U pow@@GLIBC_2.29
                 U pthread_getname_np@@GLIBC_2.12
                 U pthread_setname_np@@GLIBC_2.12
```

Particularly, the new version will no longer run on Ubuntu 18.04, potentially requiring your users to upgrade to a newer Linux release (depending on the distribution).

So probably this needs to be mentioned somewhere in the release notes.

**Reply by saudet** [reply]
It's mentioned in the README.md file already that release binaries are built against Ubuntu, but 18.04 reached EOL, so obviously that's not something that anyone should be using anyway, right?

**Reply by 0x6675636b796f75676974687562** [reply]
Fair enough, thank you!
